File: 윤석주/day2/체육복.py
def solution(n, lost, reserve):
    answer = n
    #여유가 있는 학생들의 번호 복사본
    reserveTemp = reserve.copy()
    for l in reserveTemp:
        if l in lost:
            reserve.remove(l)
            lost.remove(l)
    
    #잃어버린 학생들의 번호 복사본
    tempLost = lost.copy()
    
    for l in tempLost:
        if l-1 in reserve:
            lost.remove(l)
            reserve.remove(l-1)
        elif l+1 in reserve:
            lost.remove(l)
            reserve.remove(l+1)
    
    answer -= len(lost)
    
    return answer

File: 조재휘/23-06-2020/체육복.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    bool *take = new bool[n + 1];
    for (int i = 0; i <= n; ++i) {
        take[i] = true;
    }
    for (vector<int>::iterator iter = lost.begin(); iter != lost.end(); ++iter) {
        take[*iter] = false;
    }
    for (vector<int>::iterator iter = reserve.begin(); iter != reserve.end();) {
        if (take[*iter] == false) {
            take[*iter] = true;
            iter = reserve.erase(iter);
        }
        else {
            ++iter;
        }
    }
    for (vector<int>::iterator iter = reserve.begin(); iter != reserve.end(); ++iter) {
        if ((*iter > 1) && (take[*iter - 1] == false)) {
            take[*iter - 1] = true;
            continue;
        }
        if ((*iter < n) && (take[*iter + 1] == false)) {
            take[*iter + 1] = true;
        }
    }
    for (int i = 1; i <= n; ++i) {
        if (take[i]) {
            ++answer;
        }
    }
    return answer;
}
